File: client/src/api/baskets/BasketDto.ts
export interface BasketDto {
  userId: string;
  restaurantId: string;
  items: BasketItemDto[];
}

export interface BasketItemDto {
  menuItemId: string;
  menuItemName: string;
  menuItemDescription: string;
  menuItemPrice: number;
  quantity: number;
}

File: server/src/Web/Features/Baskets/BasketDto.cs
using System;
using System.Collections.Generic;

namespace Web.Features.Baskets
{
    public record BasketDto
    {
        public Guid UserId { get; init; }
        public Guid RestaurantId { get; init; }
        public List<BasketItemDto> Items { get; init; } = new();
    }

    public record BasketItemDto
    {
        public Guid MenuItemId { get; init; }
        public string MenuItemName { get; init; }
        public string MenuItemDescription { get; init; }
        public decimal MenuItemPrice { get; init; }
        public int Quantity { get; init; }
    }
}
